### PR TITLE
Added preprocessing files

### LINKED ISSUE #6: Create preprocessing pipeline
This pipeline will prepare data for training. Notably:

- Segment the spinal cord to extract ROI for cropping
  - Pros: Smaller input volume for training keeping only the spinal cord region in focus. Saves a lot of memory while allowing the possibility of having relatively bigger models.
- Resample to 0.5mm isotropic resolution
  - Pros: will accommodate multiple input resolutions (eg from other domains)
  - Cons: upsampling along directions where resolution is poor (eg: 3mm slice thickness --> 0.5mm)

Get started from the preprocessing here: https://github.com/ivadomed/model_seg_ms_mp2rage
EDIT: Added a Pro supporting SC segmentation

### PR DESCRIPTION
**Pull request by jcohenadad**
Fixes #6

### CODE REVIEW

File: preprocessing/config.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+########################################################################################################
```
Comment (jcohenadad) [inline preprocessing/config.yml:1]: i would advise against uploading a config file for sct_run_batch-- there is hardcoded path in it-- let's just stick to the CLI, the syntax is easy enough and it is explained in the README. 

see eg how it's done in:
- https://github.com/sct-pipeline/gm-challenge#getting-started
- https://github.com/ivadomed/model_seg_ms_mp2rage#prepare-the-data

File: preprocessing/preprocess_data.sh [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+# Preprocess data.
+#
+# Dependencies (versions):
+# - SCT (5.4.0)
+#
+# Usage:
+#   ./preprocess_data.sh <SUBJECT> <TASK>
+#
+# <SUBJECT> is the name of the subject in BIDS convention (sub-XXX)
+# <TASK> is the aimed training task which will guide preprocessing (scseg or lesionseg)
```
Comment (jcohenadad) [inline preprocessing/preprocess_data.sh:12]: that's no more needed

File: preprocessing/config.yml (hunk 1 above)
Comment (kiristern) [reply preprocessing/config.yml]: If we are no longer using a config file, the Note at the bottom of the page of [Running the sample script...](https://spinalcordtoolbox.com/user_section/tutorials/analysis-pipelines-with-sct/sct_run_batch-usage.html) tutorial should perhaps be modified.

Comment (jcohenadad) [reply preprocessing/config.yml]: @kiristern thank you for chipping in!  The note at the end of [this page](https://spinalcordtoolbox.com/user_section/tutorials/analysis-pipelines-with-sct/sct_run_batch-usage.html) refers to SCT's usage, while I was specifically referring to not using a config file for this repository only (ie: SCT users can still use a config file if they want to, but this discussion was specifically about removing the config file when running the preprocessing script to prepare the data and instead document the syntax directly in the README).

**Pull request merged**